torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4893,6 +4893,9 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            # This addmm OpInfo is for when alpha and beta are not both equal to 1.
            # alpha=beta=1 is tested in the following opinfo, because that special case will
            # trigger addmm being decomposed by a jit pass.
+           # dtypes of beta and alpha should not affect the output type, hence the type conversion
+           ref=lambda M, mat_x, mat_y, beta, alpha: np.add(np.multiply(np.asarray(beta, dtype=M.dtype), M),
```

Comment:
Please see review comments below (for `addmv`) on why dtype conversion + use of `matmul` vs `dot`.